src/shared/components/Button/Button.style.tsx

```diff
@@ -185,6 +187,32 @@ export const StyledButton = styled.button<ButtonStyleProps>`
   display: inline-flex;
   justify-content: center;
   align-items: center;
+`
+
+const LinkWithButtonProps = ({
+  variant,
+  full,
+  size,
+  disabled,
+  hasText,
+  clickable,
+  ...props
+}: ButtonStyleProps & LinkProps) => <Link {...props} />
+
+export const StyledButton = styled.button<ButtonStyleProps>`
+  ${sharedStyles}
+  &:hover {
+    cursor: ${(props) => (!props.disabled && props.clickable ? 'pointer' : '')};
+  }
+
+  ${colorsFromProps};
+  ${sizeFromProps};
+  ${disabled};
+`
+
+export const StyledLink = styled(LinkWithButtonProps)`
```

Comment:
Unfortunately, I'm still getting this: 
``` React does not recognize the `hasText` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `hastext` instead. If you accidentally passed it from a parent component, remove it from the DOM element.``` 
Trying to find a solution for this. Maybe you have a suggestion or idea on how to fix it.

Reply:
Fixed with https://emotion.sh/docs/styled#customizing-prop-forwarding